### PR TITLE
Add options to select/unselect all files

### PR DESCRIPTION
**Pull request by guanglinn**
Add options to select/unselect all files, now do some file operations more conveniently in Markor.

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -63,6 +63,7 @@ work. If not, see <https://creativecommons.org/publicdomain/zero/1.0/>.
     <string name="last_modified">Last modified</string>
     <string name="select_this_folder">Select this folder</string>
     <string name="select">Select</string>
+    <string name="select_all">Select all</string>
```
Comment (harshad1) [inline app/src/main/res/values/strings.xml:66]: IMO the button title should change to 'clear' if all items are selected.

**Comment by harshad1** [comment]
I added several related improvements in #2176 (currently open PR). These changes are
1. Refreshing the files list by swiping no longer clears the selected items.
2. Made several improvements to how we handle the selection - cleaner logic and flow etc.
3. I added a clear button to clear the selection.

This PR is more complete. So I think it makes sense for me to remove the clear button I added and instead use this. I will keep my underlying changes.

@guanglinn @gsantner I think this should be merged in first and then I will make the requisite changes to my PR

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (harshad1) [reply app/src/main/res/values/strings.xml:66]: Better yet we can do the following:

1. If no items selected make the text `Select all`
2. If one or more items selected make the text `Clear (n)` where n is the number of items selected

Then we don't need to modify the title to add a clear button etc. Changes become much simpler

**Comment by guanglinn** [comment]
> I think this should be merged in first and then I will make the requisite changes to my PR.

I agree with you, doing so can be better to improve this feature.

File: app/src/main/java/net/gsantner/opoc/frontend/filebrowser/GsFileBrowserFragment.java [hunk 2]
```diff
@@ -208,6 +216,23 @@ public void onFsViewerDoUiUpdate(GsFileBrowserListAdapter adapter) {
...
+                _topNavSubTitle.setVisibility(View.VISIBLE);
+            }
+            if (_topNavCancel.getVisibility() != View.VISIBLE) {
+                _topNavCancel.setVisibility(View.VISIBLE);
+            }
+        } else {
+            _topNavSubTitle.setText("");
+            _topNavSubTitle.setVisibility(View.GONE);
+            _topNavCancel.setVisibility(View.GONE);
+        }
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/opoc/frontend/filebrowser/GsFileBrowserFragment.java:234]: const bool isFileSelected = ...size() > 0;

if (isFileSelected) {
  _topNavSubTitle.setText(getResources().getString(R.string.selected) + " " + count;);
}
 _topNavSubTitle.setVisibility(isFileSelected? View.GONE : View.VISIBLE);
 _topNavCancel.setVisibility(isFileSelected? View.GONE : View.VISIBLE);

**Comment by gsantner** [comment]
Thank you, I'm merging this.

As noted, "clear all" is something we want too, but harshad has some things in the works anyway about the selection. So lets make this simple one as first 😄

**Comment by harshad1** [comment]
I have reworked this in my PR #2200 

Here is what it looks like now:

image

- The count is shown in the subtitle (this is built into android title bars - no need to modify title layout)
- The select all button is shown if any items selected but not all items selected
- The clear button is shown if any items selected

**Comment by guanglinn** [comment]
Nice, I should try it.

**Pull request merged**